integration-tests/client/chainlink_models.go

```diff
@@ -421,6 +421,49 @@ type SolanaNodeCreate struct {
 	Data SolanaNode `json:"data"`
 }
 
+type StarknetChainConfig struct {
+	OCR2CachePollPeriod null.String
+	OCR2CacheTTL        null.String
+	RequestTimeout      null.String
+	TxTimeout           null.Bool
+	TxSendFrequency     null.String
+	TxMaxBatchSize      null.String
+}
+
+// StarknetChainAttributes is the model that represents the starknet chain
+type StarknetChainAttributes struct {
+	Type    string              `json:"type"`
+	ChainID string              `json:"chainID"`
+	Config  StarknetChainConfig `json:"config"`
+}
+
+// StarknetChain is the model that represents the starknet chain when read
+type StarknetChain struct {
+	Attributes StarknetChainAttributes `json:"attributes"`
+}
+
+// StarknetChainCreate is the model that represents the starknet chain when created
+type StarknetChainCreate struct {
+	Data StarknetChain `json:"data"`
+}
+
+// StarknetNodeAttributes is the model that represents the starknet node
+type StarknetNodeAttributes struct {
+	Name    string `json:"name"`
+	ChainID string `json:"chainId"`
+	Url     string `json:"url"`
+}
+
+// StarknetNode is the model that represents the starknet node when read
+type StarknetNode struct {
+	Attributes StarknetNodeAttributes `json:"attributes"`
+}
+
+// StarknetNodeCreate is the model that represents the starknet node when created
+type StarknetNodeCreate struct {
+	Data StarknetNode `json:"data"`
+}
```

Comment:
Can any of these structs be made generic to chains? I see some similarity to Solana chain related structs.

Reply:
Yeah it would be much cleaner to import the existing types directly, some of which are already using generics.
Additionally, chain & node configuration is being simplified and pushed in to the general config as part of an overhaul, so we may not want to invest too much in the old style API

Reply:
I agree with the above, we should refactor and make them generic, can we please merge this so I get unblocked for StarkNET and we can tackle this as part of the ongoing refactor

Reply:
Lets merge this to unblock Starknet work and this should be refactored together with everything else when we are making generic types

Reply:
Agreed, a cleanup would be nice but let's defer it for now to unblock StarkNet 🙏

Reply:
Ideally, we make them not generic but generated =) But I love generics idea too